src/main/java/com/example/spring/beanpostprocessor/ProfilerAnnotationBeanPostProcessor.java

```diff
@@ -0,0 +1,84 @@
+package com.example.spring.beanpostprocessor;
+
+import com.example.spring.annotation.Profiling;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.stereotype.Component;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+@Component
+public class ProfilerAnnotationBeanPostProcessor implements BeanPostProcessor
+{
+   private static final Logger LOGGER =
+            LoggerFactory.getLogger(ProfilerAnnotationBeanPostProcessor.class);
+
+   @Value("${profiling.enabled}")
+   private boolean profilingEnabled;
```

Comment:
I would prefer to be able to switch profiling on runtime. There was a video with MBean implementation, so you can check